src/main/java/co/cask/hydrator/action/ftp/FTPCopyAction.java

```diff
@@ -58,79 +55,35 @@ public FTPCopyAction(FTPCopyActionConfig config) {
   /**
    * Configurations for the FTP copy action plugin.
    */
-  public class FTPCopyActionConfig extends PluginConfig {
-    @Description("Host name of the FTP server.")
-    @Macro
-    public String host;
-
-    @Description("Port on which FTP server is running. Defaults to 21.")
-    @Nullable
-    @Macro
-    public String port;
-
-    @Description("Name of the user used to login to FTP server. Defaults to 'anonymous'.")
-    @Nullable
-    @Macro
-    public String userName;
-
-    @Description("Password used to login to FTP server. Defaults to empty.")
-    @Nullable
-    @Macro
-    public String password;
+  public class FTPCopyActionConfig extends FTPActionConfig {
 
     @Description("Directory on the FTP server which is to be copied.")
     @Macro
     public String srcDirectory;
 
-    @Description("Destination directory to which the files to be copied. If the directory does not exist," +
-      " it will be created.")
-    @Macro
-    public String destDirectory;
-
     @Description("Boolean flag to determine whether zip files on the FTP server need to be extracted " +
       "on the destination while copying. Defaults to 'true'.")
     @Nullable
     public Boolean extractZipFiles;
 
-    @Description("Regex to copy only the file names that match. By default, all files will be copied.")
-    @Nullable
-    public String fileNameRegex;
-
-    public String getHost() {
-      return host;
+    public FTPCopyActionConfig(String host, String port, String userName, String password, String srcDirectory,
+                               String destDirectory, String fileNameRegex, boolean extractZipFiles) {
+      super(host, port, userName, password, destDirectory, fileNameRegex);
+      this.srcDirectory = srcDirectory;
+      this.extractZipFiles = extractZipFiles;
     }
 
     public String getSrcDirectory() {
       return srcDirectory;
     }
 
-    public String getDestDirectory() {
-      return destDirectory;
-    }
-
-    public int getPort() {
-      return (port != null) ? Integer.parseInt(port) : 21;
-    }
-
     public String getProtocol() {
```

Comment:
not added in this PR.. but this method is not used anymore I think and can be removed.